src/main.cpp

```diff
@@ -236,7 +236,15 @@ int main(int argc, char** argv)
     if (!files.isEmpty()) {
         auto file = files.constFirst();
         if (QFileInfo(file).isDir()) {
-            file.append(QLatin1String("/perf.data"));
+            // search for common perf data files in dir and open the first one found
+            // if none is found, hotspot will complain, that `file` is not a file
+            for (const auto& perfDataFile : {QStringLiteral("/perf.data"), QStringLiteral("/perf.data.old"),
+                                             QStringLiteral("/perf.data.perfparser")}) {
+                if (QFileInfo::exists(file + perfDataFile)) {
+                    file.append(perfDataFile);
+                    break;
+                }
+            }
```

Comment:
Can you move that out to a helper function that returns a QString() with the filename?
This can then also be adjusted to use the newest matching file...